pkg/resources/view.go

```diff
@@ -52,172 +59,550 @@ var viewSchema = map[string]*schema.Schema{
+	cleanupWarehouse, err := ensureWarehouse(ctx, client)
+	if err != nil {
+		return err
+	}
+	defer cleanupWarehouse()
+
+	policyRefs, err := client.PolicyReferences.GetForEntity(ctx, sdk.NewGetForEntityPolicyReferenceRequest(id, sdk.PolicyEntityDomainView))
+	if err != nil {
+		return err
+	}
+	for _, p := range policyRefs {
+		policyName := sdk.NewSchemaObjectIdentifier(*p.PolicyDb, *p.PolicySchema, p.PolicyName)
+		switch p.PolicyKind {
+		case string(sdk.PolicyKindAggregationPolicy):
+			var entityKey []string
+			if p.RefArgColumnNames != nil {
+				entityKey = sdk.ParseCommaSeparatedStringArray(*p.RefArgColumnNames, true)
+			}
+			if err = d.Set("aggregation_policy", []map[string]any{{
+				"policy_name": policyName.FullyQualifiedName(),
+				"entity_key":  entityKey,
+			}}); err != nil {
+				return err
+			}
+		case string(sdk.PolicyKindRowAccessPolicy):
+			var on []string
+			if p.RefArgColumnNames != nil {
+				on = sdk.ParseCommaSeparatedStringArray(*p.RefArgColumnNames, true)
+			}
+			if err = d.Set("row_access_policy", []map[string]any{{
+				"policy_name": policyName.FullyQualifiedName(),
+				"on":          on,
+			}}); err != nil {
+				return err
+			}
+		default:
+			log.Printf("[WARN] unexpected policy kind %v in policy references returned from Snowflake", p.PolicyKind)
+		}
+	}
+	return err
+}
+
+type viewWithReplaceDataResourceGetter struct {
+	d *schema.ResourceData
+}
+
+func (g *viewWithReplaceDataResourceGetter) Get(name string) any {
+	old, new := g.d.GetChange(name)
+	if name == "statement" {
+		return new
+	}
+	return old
 }
 
-// UpdateView implements schema.UpdateFunc.
-func UpdateView(d *schema.ResourceData, meta interface{}) error {
+func UpdateView(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	client := meta.(*provider.Context).Client
-	ctx := context.Background()
 	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
 
-	// The only way to update the statement field in a view is to perform create or replace with the new statement.
-	// In case of any statement change, create or replace will be performed with all the old parameters, except statement
-	// and copy grants (which is always set to true to keep the permissions from the previous state).
+	// change on this field can not be ForceNew because then view is dropped explicitly and copying grants does not have effect
 	if d.HasChange("statement") {
-		oldIsSecure, _ := d.GetChange("is_secure")
-		oldComment, _ := d.GetChange("comment")
-		oldTags, _ := d.GetChange("tag")
-
-		createRequest := sdk.NewCreateViewRequest(id, d.Get("statement").(string)).
-			WithOrReplace(true).
-			WithCopyGrants(true).
-			WithComment(oldComment.(string)).
-			WithTag(getTagsFromList(oldTags.([]any)))
-
-		if oldIsSecure.(bool) {
-			createRequest.WithSecure(true)
+		// TODO: extract this common code with CreateView
+		req, err := prepareCreateRequest(&viewWithReplaceDataResourceGetter{d})
```

Comment:
why do you extract the prepare create request instead of just falling back to parametrzied create?

Reply:
Because of the reasons mentioned above, changed to a parametrized func.